agent/mibgroup/if-mib/data_access/interface.c

```diff
@@ -857,3 +899,61 @@ _free_interface_config(void)
     }
     conf_list = NULL;
 }
+
+/**---------------------------------------------------------------------*/
+/*
+ * include interface config token
+ */
+/**
+ */
+static void
+_parse_include_if_config(const char *token, char *cptr)
+{
+    netsnmp_include_if_list *if_ptr, *if_new;
+    char                   *name, *st;
+
+    name = strtok_r(cptr, " \t", &st);
+    if (!name) {
+        config_perror("Missing NAME parameter");
+        return;
+    }
+    if_ptr = include_list;
+    while (if_ptr)
+        if (strcmp(if_ptr->name, name))
+            if_ptr = if_ptr->next;
+        else
+            break;
+    if (if_ptr) {
+        config_pwarn("Duplicate include interface specification");
+        return;
+    }
+    while (name != NULL) {
+        if_new = SNMP_MALLOC_TYPEDEF(netsnmp_include_if_list);
+        if (!if_new) {
+            config_perror("Out of memory");
+            return;
+        }
+        if_new->name = strdup(name);
+        if (!if_new->name) {
+            config_perror("Out of memory");
+            free(if_new);
+            return;
+        }
+        if_new->next = include_list;
+        include_list = if_new;
+        name = strtok_r(NULL, " \t", &st);
+    }
+}
+
+static void
+_free_include_if_config(void)
+{
+    netsnmp_include_if_list   *if_ptr = include_list, *if_next;
+    while (if_ptr) {
+        if_next = if_ptr->next;
+        free(NETSNMP_REMOVE_CONST(char *, if_ptr->name));
```

Comment:
Is it really useful to declare the name member 'const'? It would make me happy if the above cast would disappear.

Reply:
Ack.  I'll test a change without const.